File: vertx-elasticsearch/src/main/java/fr/myprysm/vertx/elasticsearch/action/search/aggregations/bucket/TermsBucket.java
package fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket;

import io.vertx.codegen.annotations.DataObject;
import io.vertx.core.json.JsonObject;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.Accessors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@Accessors(chain = true)
@DataObject(generateConverter = true)
public class TermsBucket extends Bucket {


    private Number keyAsNumber;
    private long docCountError;

    public TermsBucket(TermsBucket other) {
        super(other);
        keyAsNumber = other.keyAsNumber;
        docCountError = other.docCountError;
    }

    /**
     * Build a new <code>TermsBucket</code> from a <code>JsonObject</code>,
     * calling parent constructor.
     *
     * @param json the <code>JsonObject</code>
     */
    public TermsBucket(JsonObject json) {
        super(json);
        TermsBucketConverter.fromJson(json, this);
    }

    /**
     * Transforms the <code>TermsBucket</code> into a <code>JsonObject</code>,
     * calling parent <code>toJson</code>.
     *
     * @return the <code>JsonObject</code>
     */
    public JsonObject toJson() {
        JsonObject json = super.toJson();
        TermsBucketConverter.toJson(this, json);
        return json;
    }
}

File: vertx-elasticsearch/src/main/kotlin/fr/myprysm/vertx/elasticsearch/kotlin/action/search/aggregations/bucket/TermsBucket.kt
package fr.myprysm.vertx.elasticsearch.kotlin.action.search.aggregations.bucket

import fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket.TermsBucket
import fr.myprysm.vertx.elasticsearch.action.search.aggregations.Aggregation

fun TermsBucket(
        aggregations: Map<String, fr.myprysm.vertx.elasticsearch.action.search.aggregations.Aggregation>? = null,
        docCount: Long? = null,
        docCountError: Long? = null,
        key: String? = null): TermsBucket = fr.myprysm.vertx.elasticsearch.action.search.aggregations.bucket.TermsBucket().apply {

    if (aggregations != null) {
        this.setAggregations(aggregations)
    }
    if (docCount != null) {
        this.setDocCount(docCount)
    }
    if (docCountError != null) {
        this.setDocCountError(docCountError)
    }
    if (key != null) {
        this.setKey(key)
    }
}

